### PR TITLE
Add makefile target to update operator data snapshot

### PR DESCRIPTION
**Pull request by jacob-delgado**
Add a makefile target to update the operator data snapshot. Had an issue where I needed to update this file (a manifest change). 
It is also likely another target should do this (maybe `make gen`?).

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -354,6 +354,10 @@ update-golden: refresh-goldens
 
 gen: mod-download-go go-gen mirror-licenses format update-crds operator-proto sync-configs-from-istiod gen-kustomize update-golden ## Update all generated code.
 
+update-operator-data-snapshot:
```
Comment (jacob-delgado) [inline Makefile.core.mk:357]: Should this be added as a dependency for another target?

Comment (howardjohn) [reply Makefile.core.mk:357]: No, the  point is that its not updated every change, so we don't cause 100s of tests to update when we change a small file in manifests/

Should this run ./operator/scripts/run_update_golden_snapshots.sh?

Comment (jacob-delgado) [reply Makefile.core.mk:357]: We could. I can't recall what I was doing, but I had a problem with `make gen-go` or a similar command failing recently. I tracked it down to the tar.gz. I don't think adding a target makes it more easily fixable by someone. I didn't notice the run_update_golden_snapshots.sh file. I'll just close out the PR.

**Comment by jacob-delgado** [comment]
Closing since there is already `operator/scripts/run_update_golden_snapshots.sh`

**Pull request closed without merging**